src/rust/catcollar/mod.rs

```diff
@@ -185,14 +192,8 @@ impl CatcollarLibOS {
         let fd: RawFd = queue.get_fd().expect("queue should have a file descriptor");
 
         // Bind underlying socket.
-        let saddr: libc::sockaddr = linux::socketaddrv4_to_sockaddr(&local);
-        match unsafe {
-            libc::bind(
-                fd,
-                &saddr as *const libc::sockaddr,
-                mem::size_of::<libc::sockaddr_in>() as libc::socklen_t,
-            )
-        } {
+        let saddr: SockAddr = linux::socketaddrv4_to_sockaddr(&local);
+        match unsafe { libc::bind(fd, &saddr as *const SockAddr, mem::size_of::<SockAddrIn>() as Socklen) } {
```

Comment:
I assume this mem::sizeof function still works the same for wrapped types?

Reply:
Yes, these wrapped types are just aliases to the network types defined in the `windows` and `libc` crates.